app/ante_gate.go

```diff
@@ -41,3 +52,61 @@ func (ad AnteSpamMigitationDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, sim
 
 	return next(ctx, tx, simulate)
 }
+
+const (
+	//------------------- FAKE ADDRESS -------------------//
+	Address1 = "pylo1sv27mzddvum3kr7yrvptt8smyurwezzkr7psaj"
+	Address2 = "pylo1qfy4qtkgshcuzlds5z8l4mxqxc5ytdcdgr0sq7"
+	Address3 = "pylo1r84qxl40smy5tkzjhx0w6xe758qkhtgs44u5ft"
+	Address4 = "pylo1yhd2wvh4nfg4e3htzvhsq09d8wyg6eme6nte7p"
+	Address5 = "pylo1c5j57tx5m340kfn6sn99y694c5n3swjddn2rax"
+	Address6 = "pylo1hr0k73dxc53fmexxmg84qvjeva5674hkrpcs7h"
+	Address7 = "pylo12ul2m8k52clrah6r09zcrvvu0wtwkwuatmufgd"
+	Address8 = "pylo1edtuu739j3q794uwgffddehx84yhhyxd7wsjly"
+	//------------------- FAKE ADDRESS -------------------//
+)
+
+var (
+	Accounts = []string{
+		Address1,
+		Address2,
+		Address3,
+		Address4,
+		Address5,
+		Address6,
+		Address7,
+		Address8,
+	}
+)
+
+// AnteDecorator
+func (ad AnteRestrictUbedrockDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	if (ctx.IsCheckTx() || ctx.IsReCheckTx()) && !simulate {
+		sigTx, ok := tx.(authsigning.SigVerifiableTx)
+		if !ok {
+			return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "invalid transaction type")
+		}
+
+		messages := sigTx.GetMsgs()
+		fmt.Printf("[LOG] AnteRestrictUbedrockDecorator - messages : %v\n", messages)
+		for _, message := range messages {
+			msgSend, ok := message.(*banktypes.MsgSend)
+
+			sendAddr := msgSend.FromAddress
+			toAddr := msgSend.ToAddress
+
+			if ok {
+				for _, coin := range msgSend.Amount {
```

Comment:
try to use AmountOf(denom)

Reply:
Fix in https://github.com/Pylons-tech/pylons/pull/1018/commits/cc27b638812276cd04a19be78d0ab801a62b6fd2